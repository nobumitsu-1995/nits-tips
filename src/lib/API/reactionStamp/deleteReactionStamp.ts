import { customFetch } from './fetch'

export const deleteReactionStamp = async (id: number) => {
  const response = await customFetch<undefined, undefined>({
    method: 'DELETE',
    url: `/reactionStamps/${id}`,
  })

  if (!response.ok) {
    throw new Error(response.error)
  }

  return {
    ok: response.ok,
    data: 'success',
  }
}
