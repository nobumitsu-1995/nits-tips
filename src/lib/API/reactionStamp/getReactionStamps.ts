import type { StampId } from '@/components/Organisms/ReactionStamp/model'
import { customFetch } from './fetch'

type Response = {
  ReactionStampSummary: {
    StampId: StampId
    Total: number
  }[]
  ReactedStamp: {
    id: number
    stamp_id: number
  }
}

export const getReactionStamps = async () => {
  const response = await customFetch<undefined, Response>({
    method: 'GET',
    url: '/reactionStamps',
  })

  if (!response.ok) {
    throw new Error(response.error)
  }

  return response.data
}
