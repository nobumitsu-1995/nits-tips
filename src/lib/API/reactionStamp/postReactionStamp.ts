import type { StampId } from '@/components/Organisms/ReactionStamp/model'
import { fetchWithRetry } from './fetch'

type Arguments = {
  stamp_id: StampId
  article_id: string
}

type Response = {
  ID: number
  StampId: number
}

export const postReactionStamp = async (args: Arguments) => {
  const response = await fetchWithRetry<Arguments, Response>({
    method: 'POST',
    url: '/reactionStamps',
    body: args,
  })

  if (!response.ok) {
    throw new Error(response.error)
  }

  return response.data
}
