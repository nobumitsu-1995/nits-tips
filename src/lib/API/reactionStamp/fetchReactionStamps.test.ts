import { deleteReactionStamp } from './deleteReactionStamp'
import { getReactionStamps } from './getReactionStamps'
import { postReactionStamp } from './postReactionStamp'

const ARTICLE_ID = 'articleId'
test('reactionStamp取得APIを叩いて、正常に値が取得できる。', async () => {
  const response = await getReactionStamps({ articleId: ARTICLE_ID })
  expect(response).toMatchSnapshot()
})

test('reactionStamp追加APIを叩いて、正常に値が取得できる。', async () => {
  const response = await postReactionStamp({
    stamp_id: 1,
    article_id: 'hoge',
  })
  expect(response).toMatchSnapshot()
})

test('reactionStamp削除APIを叩いて、正常に処理が実行できる。', async () => {
  const response = await deleteReactionStamp(1)
  expect(response).toMatchSnapshot()
})
