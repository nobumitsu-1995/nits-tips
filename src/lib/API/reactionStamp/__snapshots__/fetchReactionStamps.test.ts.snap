// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`reactionStamp削除APIを叩いて、正常に処理が実行できる。 1`] = `
{
  "data": "success",
  "ok": true,
}
`;

exports[`reactionStamp取得APIを叩いて、正常に値が取得できる。 1`] = `
{
  "ReactedStamp": {
    "id": 18,
    "stamp_id": 1,
  },
  "ReactionStampSummary": [
    {
      "StampId": 1,
      "Total": 40,
    },
    {
      "StampId": 2,
      "Total": 7,
    },
    {
      "StampId": 3,
      "Total": 0,
    },
    {
      "StampId": 4,
      "Total": 1,
    },
    {
      "StampId": 5,
      "Total": 0,
    },
    {
      "StampId": 6,
      "Total": 0,
    },
  ],
}
`;

exports[`reactionStamp追加APIを叩いて、正常に値が取得できる。 1`] = `
{
  "ID": 68,
  "StampId": 2,
}
`;
